pkg/virt-controller/watch/dra/dra.go

```diff
@@ -584,10 +588,19 @@ func (c *DRAStatusController) getAllocatedDevice(resourceClaimNamespace, resourc
 	if resourceClaim.Status.Allocation == nil {
 		return nil, nil
 	}
-	if resourceClaim.Status.Allocation.Devices.Results == nil {
+	if len(resourceClaim.Status.Allocation.Devices.Results) == 0 {
 		return nil, nil
 	}
 
+	if requestName == "" {
+		numDevices := len(resourceClaim.Status.Allocation.Devices.Results)
+		if numDevices == 1 {
+			return resourceClaim.Status.Allocation.Devices.Results[0].DeepCopy(), nil
+		} else {
+			return nil, fmt.Errorf("requestName is required when ResourceClaim %s has multiple devices (%d devices allocated)", key, numDevices)
+		}
+	}
+
```

Comment:
I'm not familiar with this code so just commenting without that context, but with this you are establishing a new API contract, that no request means single device no matter what, I do not know if that is what you want but looks like a strong assumption, specially for DRA that have a lot of flexibility